RAGchain/schema/passage.py

```diff
@@ -107,7 +107,7 @@ def from_documents(cls, documents: List[Document]) -> List['Passage']:
                 raise ValueError(f"source must be provided in metadata, but got {metadata_etc}")
 
             previous_passage_id = ids[i - 1] if i > 0 else None
-            next_passage_id = ids[i + 1] if i < len(documents) - 1 else None
+            next_passage_id = ids[i + 1] if i < len(ids) - 1 else None
```

Comment:
change len(documents) to len(ids) 
because error occurs when len(documents) == 1 or 2.

The same error occurs in the from_search function below, which I fixed (last commit)